src/backend/core/services/proconnect.py

```
@@ -0,0 +1,752 @@
+    canonically. This keeps equality checks (change detection, spurious-write
+    avoidance) order- and duplicate-insensitive everywhere they are compared.
+
+    Only hostname-charset values (``[a-z0-9.-]``) are kept: this drops anything with
+    whitespace, ``#`` or ``/`` before it can reach the public allowlist YAML (built
+    by string concatenation), preventing YAML injection via a superuser-set domain.
+    """
+    cleaned = set()
+    for domain in domains or []:
+        if not isinstance(domain, str):
+            continue
+        domain = domain.strip().lower()
```

Ruff (0.16.0)

[warning] `for` loop variable `domain` overwritten by assignment target

(PLW2901)